modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/display/context/DLCopyFolderDisplayContext.java

```diff
@@ -0,0 +1,183 @@
+		ItemSelector itemSelector =
+			(ItemSelector)_httpServletRequest.getAttribute(
+				ItemSelector.class.getName());
+
+		return String.valueOf(
+			itemSelector.getItemSelectorURL(
+				RequestBackedPortletURLFactoryUtil.create(_httpServletRequest),
+				_getGroup(getSourceRepositoryId()),
+				_themeDisplay.getScopeGroupId(), _getItemSelectedEventName(),
+				_getFolderItemSelectorCriterion(
+					getSourceFolderId(), getSourceRepositoryId())));
+	}
+
+	public long getSourceFolderId() {
+		if (_sourceFolderId < 0) {
+			_sourceFolderId = ParamUtil.getLong(
+				_httpServletRequest, "sourceFolderId");
+		}
+
+		return _sourceFolderId;
+	}
+
+	public String getSourceFolderName() {
+		if (_sourceFolderName != null) {
+			return _sourceFolderName;
+		}
+
+		_sourceFolderName = "Home";
+
+		DLFolder folder = DLFolderLocalServiceUtil.fetchFolder(
+			getSourceFolderId());
+
+		if (folder != null) {
+			_sourceFolderName = folder.getName();
+		}
+
+		return _sourceFolderName;
+	}
+
+	public long getSourceRepositoryId() {
+		if (_sourceRepositoryId < 0) {
+			_sourceRepositoryId = ParamUtil.getLong(
+				_httpServletRequest, "sourceRepositoryId");
+		}
+
+		return _sourceRepositoryId;
+	}
+
+	private FolderItemSelectorCriterion _getFolderItemSelectorCriterion(
+		long folderId, long repositoryId) {
+
+		FolderItemSelectorCriterion folderItemSelectorCriterion =
+			new FolderItemSelectorCriterion();
+
+		folderItemSelectorCriterion.setDesiredItemSelectorReturnTypes(
+			new FolderItemSelectorReturnType());
+		folderItemSelectorCriterion.setFolderId(folderId);
+		folderItemSelectorCriterion.setIgnoreRootFolder(true);
+		folderItemSelectorCriterion.setRepositoryId(repositoryId);
+		folderItemSelectorCriterion.setSelectedFolderId(folderId);
+		folderItemSelectorCriterion.setSelectedRepositoryId(repositoryId);
+		folderItemSelectorCriterion.setShowGroupSelector(true);
+		folderItemSelectorCriterion.setShowMountFolder(false);
+
+		return folderItemSelectorCriterion;
+	}
+
+	private Group _getGroup(long repositoryId) throws PortalException {
+		long groupId = repositoryId;
+
+		Repository repository = RepositoryLocalServiceUtil.fetchRepository(
+			repositoryId);
+
+		if (repository != null) {
+			groupId = repository.getGroupId();
+		}
+
+		return GroupLocalServiceUtil.getGroup(groupId);
+	}
+
+	private String _getItemSelectedEventName() {
+		PortletDisplay portletDisplay = _themeDisplay.getPortletDisplay();
+
+		return portletDisplay.getNamespace() + "folderSelected";
+	}
+
+	private void _setViewAttributes() {
+		PortletDisplay portletDisplay = _themeDisplay.getPortletDisplay();
+
+		portletDisplay.setShowBackIcon(true);
+		portletDisplay.setURLBack(getRedirect());
+
+		if (_liferayPortletResponse instanceof RenderResponse) {
+			RenderResponse renderResponse =
+				(RenderResponse)_liferayPortletResponse;
+
+			renderResponse.setTitle(
+				LanguageUtil.get(_httpServletRequest, "copy-to"));
+		}
+	}
```

Comment:
I don't think mutating the request in the constructor is a good idea. I'd either leave the code in the JSP (it is ugly, but it is the traditional way), or expose this method and call it explicitly from the JSP.

Reply:
I also didn't like the approach I took. I copied it from another place, I supposed it was a "standard" 😓 Now I've exposed the method `setViewAttributes`. I've also changed the required parameters for the `DLCopyFolderDisplayContext` constructor.

See the commit [LPS-182876 minor improvements](https://github.com/liferay-lima/liferay-portal/pull/4281/commits/2536104f0d5201b0990dea10509f0725a37a8f21)